### PR TITLE
Add a remote-close integration test.

### PR DESCRIPTION
**Pull request by mollyling**
Yeah, just:
-  setup to listen to a close event locally, 
-  tell the echo server to close its connections
- then make sure we got the close event.

[Review on Reviewable](https://reviewable.io/reviews/uproxy/uproxy-networking/226)

### CODE REVIEW

**Comment by mollyling** [comment]
The continuous build failure is on: "Warning: Unable to read "/home/travis/build/uProxy/uproxy-networking/node_modules/freedom-pgp-e2e/build/*_/_" file (Error code: ENOENT). Use --force to continue.".  Which frankly, befuddles me.  Should I rebase off of a newer branch to fix?

**Comment by iislucas** [comment]
Yay! More integration tests!

That's probably just an old `freedom-pgp-e2e` library in node_modules; nuke the node modules and re-install. 

Also: you probably want to redo this pull request against dev (dev requirefiy is now merged into dev)

File: src/integration-tests/socks-echo/base-spec.core-env.ts [hunk 1]
```diff
@@ -54,6 +54,28 @@ export function socksEchoTestDescription(useChurn:boolean) {
     }).then(done);
   });
 
+  it('detects a remote close', (done) => {
+    var input = arraybuffers.stringToArrayBuffer('arbitrary test string');
+    var testModule = getTestModule();
+    var connid : string;
```
Comment (trevj) [inline src/integration-tests/socks-echo/base-spec.core-env.ts]: super nit:

``` typescript
var connId :string;
```

File: src/integration-tests/socks-echo/base-spec.core-env.ts [hunk 2]
```diff
@@ -54,6 +54,28 @@ export function socksEchoTestDescription(useChurn:boolean) {
...
+    var testModuleCopy:any;
```
Comment (trevj) [inline src/integration-tests/socks-echo/base-spec.core-env.ts]: This is weird, I don't see where it's used.

**Comment by trevj** [comment]
Again, it would be good to rebase this off of the `dev` branch. Thanks again for adding this!

**Comment by mollyling** [comment]
<sup>**[src/integration-tests/socks-echo/base-spec.core-env.ts, line 60 [r1]](https://reviewable.io:443/reviews/uproxy/uproxy-networking/226#-JmKV8fVS9IhPSFR8mYC-r1-60)** ([raw file](https://github.com/uproxy/uproxy-networking/blob/21bfe3bb2727e8c6b86b7233382b175a7d02559b/src/integration-tests/socks-echo/base-spec.core-env.ts#L60)):</sup>
Done.

---

<sup>**[src/integration-tests/socks-echo/base-spec.core-env.ts, line 61 [r1]](https://reviewable.io:443/reviews/uproxy/uproxy-networking/226#-JmKV8fVS9IhPSFR8mYC-r1-61)** ([raw file](https://github.com/uproxy/uproxy-networking/blob/21bfe3bb2727e8c6b86b7233382b175a7d02559b/src/integration-tests/socks-echo/base-spec.core-env.ts#L61)):</sup>
Done.

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy-networking/226)

**Comment by mollyling** [comment]
Rebased to dev.

**Comment by mollyling** [comment]
This has gone somewhere awful.  Killing it.

**Pull request closed without merging**